### PR TITLE
[DataPipe] Small change to generation script and update to DataPipe .pyi file

### PR DESCRIPTION
**Pull request by NivekT**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #69392
* #69393
* #69391
* #69390

cc @VitalyFedyunin @ejguan @NivekT

Differential Revision: [D32849463](https://our.internmc.facebook.com/intern/diff/D32849463)

### CODE REVIEW

File: torch/utils/data/dataset.pyi [hunk 1]
```diff
@@ -42,19 +44,33 @@ class IterableDataset(Dataset[T_co], metaclass=_DataPipeMeta):
...
+    # Functional form of 'ConcaterIterDataPipe'
     def concat(self, *datapipes: IterDataPipe): ...
+    # Functional form of 'RoutedDecoderIterDataPipe'
     def decode(self, *handlers: Callable, key_fn: Callable= ...): ...
-    def demux(self, num_instances: int, classifier_fn: Callable[[T_co], int], drop_none: bool = False, buffer_size: int = 1000): ...
-    def filter(self, filter_fn: Callable, fn_args: Optional[Tuple] = None, fn_kwargs: Optional[Dict] = None, drop_empty_batches: bool = True, nesting_level: int = 0): ...
+    # Functional form of 'DemultiplexerIterDataPipe'
+    def demux(self, num_instances: int, classifier_fn: Callable[[T_co], Optional[int]], drop_none: bool = False, buffer_size: int = 1000): ...
+    # Functional form of 'FilterIterDataPipe'
+    def filter(self, filter_fn: Callable, fn_args: Optional[Tuple] = None, fn_kwargs: Optional[Dict] = None, drop_empty_batches: bool = True): ...
```
Comment (ejguan) [inline torch/utils/data/dataset.pyi:58]: Can you move the change of `map` and `filter` API to corresponding PRs?

**Review by ejguan — APPROVED** [review]
LGTM except one comment.

File: torch/utils/data/dataset.pyi (hunk 1 above)
Comment (NivekT) [reply torch/utils/data/dataset.pyi:58]: Done!

**Comment by NivekT** [comment]
@nivekt has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32849463).

**Review by ejguan — COMMENTED** [review]
BTW, can we at least make the functional APIs within pytorch core being dynamically added into pyi file?
It's really annoying to maintain this type stub file. Can we add it to our TODO list of next half.

**Comment by NivekT** [comment]
> BTW, can we at least make the functional APIs within pytorch core being dynamically added into pyi file? It's really annoying to maintain this type stub file. Can we add it to our TODO list of next half.

Agreed, I think the `gen_pyi.py ` file should be automatically ran as part of the CI process. I will add it to the list

**Pull request closed without merging**